x/tier/keeper/portfolio.go

```diff
@@ -214,12 +216,19 @@ func (k Keeper) RetrieveRewardsTotal(ctx sdk.Context, user sdk.AccAddress) sdkma
 func (k Keeper) RetrievePerpetualTotal(ctx sdk.Context, user sdk.AccAddress) (sdkmath.LegacyDec, sdkmath.LegacyDec, sdkmath.LegacyDec) {
 	totalAssets := sdkmath.LegacyNewDec(0)
 	totalLiability := sdkmath.LegacyNewDec(0)
-	netValue := sdkmath.LegacyNewDec(0)
-	perpetuals := k.perpetual.GetAllMTPsForAddress(ctx, user)
+	var netValue sdkmath.LegacyDec
+	perpetuals, _, err := k.perpetual.GetMTPsForAddressWithPagination(ctx, user, nil)
+	if err != nil {
+		return sdkmath.LegacyNewDec(0), sdkmath.LegacyNewDec(0), sdkmath.LegacyNewDec(0)
+	}
 	for _, perpetual := range perpetuals {
-		totalAssets = totalAssets.Add(k.CalculateUSDValue(ctx, perpetual.GetTradingAsset(), perpetual.Custody))
-		totalLiability = totalLiability.Add(k.CalculateUSDValue(ctx, perpetual.LiabilitiesAsset, perpetual.Liabilities))
-		totalLiability = totalLiability.Add(k.CalculateUSDValue(ctx, perpetual.LiabilitiesAsset, perpetual.BorrowInterestUnpaidLiability))
+		if perpetual.Mtp.Position == perpetualtypes.Position_LONG {
+			totalAssets = totalAssets.Add(k.CalculateUSDValue(ctx, perpetual.Mtp.GetTradingAsset(), perpetual.Mtp.Custody))
+		} else {
+			totalAssets = totalAssets.Add(perpetual.Mtp.Custody.ToLegacyDec())
+		}
+		totalLiability = totalLiability.Add(k.CalculateUSDValue(ctx, perpetual.Mtp.LiabilitiesAsset, perpetual.Mtp.Liabilities))
+		totalLiability = totalLiability.Add(k.CalculateUSDValue(ctx, perpetual.Mtp.LiabilitiesAsset, perpetual.Mtp.BorrowInterestUnpaidLiability))
```

Comment:
We can add perpetual.Mtp.BorrowInterestUnpaidLiability + perpetual.Mtp.Liabilities do calculations in one go. Also, for Long it's always in USDC, so can skip k.CalculateUSDValue for it just like for custody?

Reply:
done